mongodb/mongo_import.cc

```diff
@@ -0,0 +1,228 @@
+        const auto namedBound  = runConfig.named->bind(mongoScope);
+
+        bool useSelect = config.select != SelectExpression::STAR;
+        bool useWhere = !config.where->isConstantTrue();
+
+        // using incorrect default value to ease check
+        bool useNamed = config.named != SqlExpression::TRUE;
+
+        auto processor = [&](Dataset & output,
+                             const bsoncxx::document::view & doc)
+        {
+            if (doc["_id"].type() != bsoncxx::type::k_oid) {
+                throw HttpReturnException(
+                    500,
+                    "monbodb.import unimplemented support for "
+                    "MongoDB records with key \"_id\" that are not "
+                    "ObjectIDs.");
+            }
+            auto oid = doc["_id"].get_oid();
+            Path rowName(oid.value.to_string());
+            auto ts = Date::fromSecondsSinceEpoch(oid.value.get_time_t());
+            ExpressionValue expr(extract(ts, doc));
+
+            if (useWhere || useSelect || useNamed) {
+                MongoRowScope row(expr, oid.value.to_string());
+                if (useWhere && !whereBound(row, storage, GET_ALL).isTrue()) {
+                    return;
+                }
+
+                if (useNamed) {
+                    rowName = RowName(
+                        namedBound(row, storage, GET_ALL).toUtf8String());
+                }
+
+                if (useSelect) {
+                    expr = selectBound(row, storage, GET_ALL);
+                    storage = expr;
+                }
+            }
+
+            output.recordRowExpr(rowName, expr);
+        };
+
+        auto offset = runConfig.offset;
+        auto limit = runConfig.limit;
+        int errors = 0;
+        {
+            auto cursor = db[runConfig.collection].find({});
+            int i = 0;
+            for (auto&& doc : cursor) {
+                if (offset > 0) {
+                    --offset;
+                    continue;
+                }
+                if (limit == 0) {
+                    break;
+                }
+                else if (limit > 0) {
+                    --limit;
+                }
+                if (++i % 1000 == 0) {
+                    DEBUG_MSG(logger) << "Processing " << i << "th document";
+                }
+                if (runConfig.ignoreParsingErrors) {
+                    try {
+                        processor(*output.get(), doc);
+                    }
+                    catch (const ML::Exception & exc) {
+                        ++ errors;
+                        logger->error() << exc.what();
+                    }
+                }
+                else {
+                    processor(*output.get(), doc);
+                }
+            }
+            DEBUG_MSG(logger) << "Fetched " << i << " documents";
+        }
+        output->commit();
+        Json::Value res = jsonEncode(output->getStatus());
```

Comment:
It would be great to have more information return.  At a minimum the number of rows processed.